tomcat-conf/Catalina/localhost/prweb.xml

```diff
@@ -32,5 +32,13 @@
   <Environment name="prconfig/session/ha/enabled" value="true" type="java.lang.String"/>
   <Environment name="prconfig/dsm/services/stream/pyBaseLogPath" value="/opt/pega/kafkadata" type="java.lang.String" />
   <Environment name="prconfig/cloud/isNodeAgnosticAdminStudio" value="true" type="java.lang.String"/>
+
+  {{ if isTrue .Env.CLIENT_MODE }}
+  <Environment name="prconfig/cluster/clientserver/clientmode" value="{{ .Env.CLIENT_MODE }}" type="java.lang.String"/>
+  <Environment name="prconfig/Pega-RULES/onPegaK8s" value="{{ .Env.ON_PEGA_K8S }}" type="java.lang.String"/>
```

Comment:
Why do we need to have a parameter that specifies architecture implementation?  Should we be naming this parameter based on the action it performs instead?

Also, this Docker image is only used on Kubernetes deployments today.

Reply:
Are you referring to prconfig name (`prconfig/Pega-RULES/onPegaK8s`)  or environment variable name (`ON_PEGA_K8S`).

We deinite;y need to pass prconfig/cluster/clientserver/clientmode in prconfig because we want to indicate that pega infinity node is starting in hazelcast client mode.

There are different types of client server model , pega cuttyhunk ,  on-premise and  K8s deployment. So to indicate that this client mode is for K8s deployment , we need to set this value ` <Environment name="prconfig/Pega-RULES/onPegaK8s" value="{{ .Env.ON_PEGA_K8S }}" type="java.lang.String"/>`  to true.

As prconfig/Pega-RULES/onPegaK8s is new setting in prweb.xml and ON_PEGA_K8S is a new environment variable , I feel there should be no problem in changing either of these names.

Reply:
What is the behavioral change in Pega when onPegak8s is true?  Is it specific to Hazelcast?

Reply:
> Why do we need to have a parameter that specifies architecture implementation? Should we be naming this parameter based on the action it performs instead?
> 
> Also, this Docker image is only used on Kubernetes deployments today.

Actually, we use this as the base image for running local instances for pega developers, so I'm worried all this new hazelcast stuff might break what we have for local dev

Reply:
It should not break. The default behavior in this image is to run hazelcast in embedded mode which happens today also.
HZ client server work is still WIP. The change we have done in this image is passing few settings in prweb.xml and then added few additional environment variables in hazelcast. They should not break any current deployment process.

Reply:
> What is the behavioral change in Pega when onPegak8s is true? Is it specific to Hazelcast?

yes its specific to hazelcast. We need this flag to determine the discovery strategy to be used. On aws , infinity makes use of a customized discovery plugin for discovering servers , in embedded we make use of tcp/ip and for kubernetes we make use of kubernetes hazelcast plugin.

Reply:
If this is Hazelcast specific, it needs a different name that ties it to the *behavior* that is impacted. For example, if this is for connecting to the docker based Hz image, you should name the flag for that.

Reply:
Yes @dcasavant , we renamed the config to "cluster/clientserver/hzOnK8s".